### DISCUSSION TITLE
Split the init.lua into more files/modules

### DISCUSSION THREAD

**Discussion by ducktordanny** [p5quared/apple-music.nvim#16] [Ideas]
As the feature set grows it would be easier to work with the code base by having it split up into more modules. I could think of something like this:

- `apple-music.controls`: play_track, pause, resume, next_track, previous_track, etc.
- `apple-music.list`: telescope and related functions
- `apple-music.details`: get_current_trackname, get_current_author, etc.
- `apple-music.utils`: internal functions maybe?
- etc.

Any thoughts on this? :)

**Comment by p5quared** [comment]
I'm not quit sure it's big enough yet but I think it's definitely something to keep in mind. Even just having a main file and a utils file (everything that is not public) would be useful.

**Reply by ducktordanny** [reply]
That already sounds good, to have the utils separate. I think I'll take a look at that.

**Reply by p5quared** [reply]
Part of the reason I don't want to change too much is because fore very structure change I may have to edit the doc generation.
